Update Transifex config format for latest version

dacook opened the pull request:
#### What? Why?

This updates the Transifex config as required by the latest version (CLI 1.4.1; APIv3)

If you have the Transifex CLI client installed, you may need to update it to the latest version:

    tx update

> Congratulations, you are up to date with  1.4.1


#### What should we test?
This can be tested in a development environment by running the following:

    tx pull

You are expecting a successful result, eg:

```
# Getting info about resources

open-food-network.enyml - Done
[##############################] (1 / 1)

# Pulling files
...
```

#### Release notes
Changelog Category: Technical changes

#### Dependencies
It doesn't depend on an OFN PR, but may require the latest version of the Transifex CLI client (see above)


#### Documentation updates
I don't think we need to update either of these:
 - https://github.com/openfoodfoundation/openfoodnetwork/wiki/Internationalisation-(i18n)#transifex-client
 - https://github.com/openfoodfoundation/openfoodnetwork/wiki/Releasing#pull-translations-in

(Although I have made some improvements to the former already)

jibees reviewed (approved):
Thanks!

jibees commented:
### Testing notes

Upgrading the `tx` CLI program (via https://developers.transifex.com/docs/cli#installing-with-a-script-linuxmac) into a folder I chose.
```
➜  openfoodnetwork git:(pr/9935) tx --version
TX Client, version=1.4.1
```
I tried `tx pull`:
```
➜  openfoodnetwork git:(pr/9935) tx pull
2022/11/04 16:05:22 could not find a Transifex API host and/or TOKEN, please inspect your .transifexrc and .tx/config files
```

But the file `.tx/config` actually exists. And I have a `$HOME/.transifexrc` file:
```
➜  openfoodnetwork git:(pr/9935) ls $HOME/.transifexrc
/Users/jibees/.transifexrc
```
Seems like someone opened an issue: https://github.com/transifex/cli/issues/119 (but closed it)

To conclude:
 - I can't make it work. I'm still wondering why.
 - We probably should update our wiki in order to give instruction to the new Transifex client (ie. https://github.com/transifex/cli instead of https://github.com/transifex/transifex-client)

mkllnk commented:
```
$ tx --version
0.14.2, py 2.7, x86_64

$ tx update
/usr/lib/python2.7/dist-packages/requests/__init__.py:91: RequestsDependencyWarning: urllib3 (1.26.4) or chardet (3.0.4) doesn't match a supported version!
  RequestsDependencyWarning)
tx ERROR: Command update not found
```

I have a very old version which doesn't support the automated update yet.

mkllnk commented:
So I installed it according to the instructions:
```
curl -o- https://raw.githubusercontent.com/transifex/cli/master/install.sh | bash
...
TX Client, version=1.4.1
```

I needed to run:
```
tx migrate
```

It updated my .transifexrc and added two lines:
```
rest_hostname = https://rest.api.transifex.com
token         = 1/7ff2cc...
```

Then I can run `tx pull --force` on this PR and it's faster than the old version. :+1: 

Switching back to master, the new old config doesn't work. So we need to merge this now. :smile:

mkllnk commented:
@jibees Try `tx update`. If that doesn't work then you can try the install script above. It will install it in your current directory. So maybe go somewhere like `$HOME/bin` where you want additional binaries installed.

dacook commented:
Thanks for the link @jibees, I've updated the wiki now with the new link. I also had that error and fixed it by changing file permissions for my `~/.transifexrc` file, so added that to the wiki too: 
https://github.com/openfoodfoundation/openfoodnetwork/wiki/Internationalisation-(i18n)
(Thanks Maikel for the suggestion).

jibees commented:
Thanks for the tip, I needed to run `tx update` as well 👍
